cmake/3rdParty/script-only-mode/PostProcessScriptOnlyMappings.cmake

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# this file is used during generation of the installer layout.
+# note that ARGV0 is the cmake command itself, and ARGV1 is the -P command line option, so we skip them
+# ARGV2 is the name of this script.
+# ARGV3 is the output folder name
+# ARGV4 and onwards are the names of input files, generated during the engine configure/generate
+# containing information about the 3p libraries avilable.
+
+#[[ there is 1 input file for each 3p library, and each input file contains a set of "set" commands in this format:
+    set(target_name ${TARGET_NAME})
+    set(copy_dependencies_source_file_path ${copy_dependencies_source_file_path})
+    set(copy_dependencies_target_rel_path ${copy_dependencies_target_rel_path})
+
+    The goal of this file is to generate one output_filename/target_name.cmake for each input file such 
+    that it contains the ability to register the given 3p library target
+    libraries as fake ones (which have the appropriate dependencies so that files get copied).
+]]#
+
+if (${CMAKE_ARGC} LESS 3)
+    message(FATAL_ERROR "This command was invoked incorrectly, expected at least the first three args to be cmake itself, then -P, then outputfilename")
```

Comment:
Not sure why this comment was hidden, is this still legit?
The caller makes a command like this
```
COMMAND ${CMAKE_COMMAND} -P 
                "${LY_ROOT_FOLDER}/cmake/3rdParty/script-only-mode/PostProcessScriptOnlyMappings.cmake" # Script to run
                "${final_3p_output_dir}" # the output file to make
                ${list_of_files_to_process}
 ```
 
 This would imply that
 ```
 ARGV0 = "Cmake"
 ARGV1 = "-P"
 ARGV2 = "this script, as in, PostProcessScriptOnlyMappings.cmake"
 ARGV3 = "${final_3p_output_dir}"
 ARGV4  and onwards are the scripts to run.
 
 This agrees with the current comment, which is 
 ```
 # this file is used during generation of the installer layout.
# note that ARGV0 is the cmake command itself, and ARGV1 is the -P command line option, so we skip them
# ARGV2 is the name of this script.
# ARGV3 is the output folder name
# ARGV4 and onwards are the names of input files, generated during the engine configure/generate
# containing information about the 3p libraries available.

```

Reply:
Well the fatal error message implies that the 3rd argument is the output file to write to
> expected at least the first three args to be cmake itself, then -P, then outputfilename

Reply:
aha, I'll update the fatal error